pkg/jsonpath/eval.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jsonpath
+
+// This file implements JSONPath querying almost as described in RFC 9535.
+// This file does not implement the function extensions.
+// The only comparisons implemented here are equality and inequality.
+// The order in which result nodes are produced is not consistent with the RFC.
+
```

Comment:
could this lead to unexpected behavior for the users expecting RFC-compliant ordering ?

Reply:
Oops, forgot to update that comment. Fixed in the latest push.